### PR TITLE
Fix to properly access bio_vecs using bi_iter via macros in read-modify-write code path

### PR DESCRIPTION
**Pull request by venkrishraja**
Replaced direct accesses to bi_io_vec with appropriate macros

### CODE REVIEW

File: dm-dedup-rw.c [hunk 1]
```diff
@@ -55,6 +55,8 @@ static int merge_data(struct dedup_config *dc, struct page *page,
 	sector_t bi_sector = bio->bi_iter.bi_sector;
 	void *src_page_vaddr, *dest_page_vaddr;
 	int position, err = 0;
+	struct bvec_iter iter;
+    struct bio_vec bvec;
```
Comment (ezk) [inline dm-dedup-rw.c:59]: Does this line have a strange indentation?

**Comment by venkrishraja** [comment]
No. It looks fine on my editor. Don't know why it shows like this in the diff.

**Comment by ezk** [comment]
It’s wrong in the patch itself.
-   struct bvec_iter iter;
-    struct bio_vec bvec;

The first line has a TAB char.  The second one has 4 spaces instead.  You should be careful: configure your text editor for “standard” linux/c editing style, which uses  TABs.  In Emacs it’s

```
CTRL-c . linux
```

(note “.” after Control-C)

Run ./scripts/checkpatch.pl with right args on EVERY patch before you submit it, and fix any warnings/errors checkpatch.pl reports.  Indentation is one of them.

Thanks,
Erez.

**Comment by venkrishraja** [comment]
Sure.

**Pull request closed without merging**